src/psyclone/tests/psyad/main_test.py

```diff
@@ -134,20 +137,89 @@ def test_main_no_filename(capsys):
     # of the executable is replaced with either pytest or -c when
     # using pytest, therefore we split the test into sections.
     expected1 = "usage: "
-    expected2 = "[-h] [-v] [-t] [-otest TEST_FILENAME] [-oad OAD] filename"
+    expected2 = ("[-h] [-oad OAD] [-v] [-t] [-otest TEST_FILENAME] "
+                 "-a ACTIVE [ACTIVE ...] -- filename")
     if six.PY2:
         expected3 = "error: too few arguments\n"
     else:
-        expected3 = "error: the following arguments are required: filename\n"
+        expected3 = ("error: the following arguments are required: "
+                     "-a/--active, filename\n")
     assert expected1 in error
     assert expected2 in error
     assert expected3 in error
 
 
+# no -a
+def test_main_no_a_arg(capsys):
+    '''Test that the main() function raises an exception if the -a
+    argument is not supplied.
+    '''
+    with pytest.raises(SystemExit) as info:
+        main(["file"])
+    assert str(info.value) == "2"
+    output, error = capsys.readouterr()
+    assert output == ""
+    expected = ("error: the following arguments are required: "
+                "-a/--active")
+    if six.PY2:
+        expected = "argument -a/--active is required"
+    assert expected in error
+
+
+# no -a arg arg
+def test_main_no_a_arg_arg(capsys):
+    '''Test that the main() function raises an exception if an argument to
+    the -a argument is not supplied.
+    '''
+    with pytest.raises(SystemExit) as info:
+        main(["file", "-a"])
+    assert str(info.value) == "2"
+    output, error = capsys.readouterr()
+    assert output == ""
+    expected = ("error: argument -a/--active: expected at least one "
+                "argument")
+    assert expected in error
+
+
+# no filename
+def test_main_no_filename(capsys):
+    '''Test that the main() function raises an exception if no filename is
+    supplied.
+    '''
+    with pytest.raises(SystemExit) as info:
+        main(["-a", "var"])
+    assert str(info.value) == "2"
+    output, error = capsys.readouterr()
+    assert output == ""
+    expected = "error: the following arguments are required: filename\n"
+    if six.PY2:
+        expected = "error: too few arguments\n"
+    assert expected in error
+
+
+# no --
+def test_main_no_separator(capsys):
+    '''Test that the main() function raises an exception if the -- is not
+    provided between the argument and the filename.
+
+    '''
+    with pytest.raises(SystemExit) as info:
+        main(["-a", "var", "filename"])
+    assert str(info.value) == "2"
+    output, error = capsys.readouterr()
+    assert output == ""
+    expected = "error: the following arguments are required: filename\n"
```

Comment:
This could be confusing. Is it possible to extend the message to say something like "...filename - perhaps you are missing the '--' after the list of active variables?"

Reply:
I agree, but as discussed this message is produced by the library handling the arguments, not by psyad.